### PR TITLE
gnomeExtensions.system-monitor: Patch for GNOME 43 compatibility

### PR DESCRIPTION
**Pull request by andersk**
###### Description of changes

gnome-shell-system-monitor needs a small patch for GNOME 43 compatibility: paradoxxxzero/gnome-shell-system-monitor-applet#754

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome/extensions/system-monitor/default.nix [hunk 1]
```diff
@@ -17,6 +17,11 @@ stdenv.mkDerivation rec {
   ];
 
   patches = [
+    # GNOME 43 compatibility
+    (fetchpatch {
+      url = "https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/pull/754/commits/3b3a617f78c5e9bbce0aa2864b3989335edb37b7.patch";
```
Comment (bjornfor) [inline pkgs/desktops/gnome/extensions/system-monitor/default.nix:22]: Won't this URL be garbage collected (breaking long-term reproducibility)?

Comment (andersk) [reply pkgs/desktops/gnome/extensions/system-monitor/default.nix:22]: I don’t see any reason to think that. https://github.com/NixOS/nixpkgs/pull/1/commits/d0d4123962ed53266319dff6b19c42444b26bb7a.patch stlil works.

Comment (bjornfor) [reply pkgs/desktops/gnome/extensions/system-monitor/default.nix:22]: That's good to know. I wonder if that holds if a PR is rebased before merging though? Does github really keep every revision, even "unreferenced" ones?

Comment (piegamesde) [reply pkgs/desktops/gnome/extensions/system-monitor/default.nix:22]: As far as I know, by default yes.

**Review by bjornfor — CHANGES_REQUESTED** [review]
https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/pull/754#discussion_r1010706826

File: pkgs/desktops/gnome/extensions/system-monitor/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/desktops/gnome/extensions/system-monitor/default.nix:22]: They keep them long enough that fetching patches like this was not a major problem yet unless the project is completely deleted.

File: pkgs/desktops/gnome/extensions/system-monitor/default.nix [hunk 2]
```diff
@@ -17,6 +17,15 @@ stdenv.mkDerivation rec {
...
+      hash = "sha256-BAD0ExQYmTg6i02Gg7jcF0vf1zp232vXRjGCn+rBYXE=";
+    })
+    (fetchpatch {
```
Comment (piegamesde) [inline pkgs/desktops/gnome/extensions/system-monitor/default.nix:25]: Could we use the entire PR as a single patch? I know that this is possible in theory, just not sure about how versioning / pinning works in that case

Comment (andersk) [reply pkgs/desktops/gnome/extensions/system-monitor/default.nix:25]: There is https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/pull/754.patch, but that’s only useful to us for a *merged* PR, as any update would invalidate its hash.

**Pull request merged**